### PR TITLE
ARROW-984: [GLib] Add Go examples

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: c_glib/example/go/write-batch.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		BuildInt8Array(),
+		BuildInt16Array(),
+		BuildInt32Array(),
+		BuildInt64Array(),
+		BuildFloatArray(),
+		BuildDoubleArray(),
+	}
+
+	recordBatch := arrow.NewRecordBatch(schema, 4, columns)
+	writer.WriteRecordBatch(recordBatch)
```
Comment (wesm) [inline c_glib/example/go/write-batch.go]: Can this return an error?

**Review by wesm — APPROVED** [review]
+1, very cool

**Comment by wesm** [comment]
I'm interested to start a native Go implementation, since this will only run on Linux, right?

File: c_glib/example/go/write-batch.go (hunk 1 above)
Comment (kou) [inline c_glib/example/go/write-batch.go]: Yes. I added error checks for them by #666.

**Comment by kou** [comment]
> I'm interested to start a native Go implementation,

It'll be useful because it'll be faster and provide more Go style API than GLib bindings based bindings. Maintaining the implementation may be a problem.

> since this will only run on Linux, right?

It's not right. The GLib bindings and bindings based on the GLib bindings can work on Linux, macOS and Windows. At least Ruby bindings can work on them. I'm developing GTK+ bindings with GObject Introspection. (gtk3 gem) The bindings can work on Linux, macOS and Windows.

But installing from source will be difficult for users. We need to provide packages for easy to use.

For macOS, it's better that we send a pull request to [Homebrew](https://github.com/Homebrew/homebrew-core).

For Windows, it's better that we build binary and make it downloadable. I'm doing it by cross-compiling on my Debian GNU/Linux machine for gtk3 gem.

**Comment by wesm** [comment]
Ah that's good to know about GLib. I would like to make a minimal implementation for Go that can read and write the streaming format and provide access to the raw buffers. I am not sure when I will have the time though =) Probably not until 2018... 

It will also need to provide binaries for integration testing. Quite a bit of work; if we can recruit some developres from the Go community (like users or developers of InfluxDB or another database written in Go) that might be a good place to start

**Pull request closed without merging**